pkg/traceql/ast_execute.go

```diff
@@ -325,30 +325,50 @@ func (a Aggregate) evaluate(input []*Spanset) (output []*Spanset, err error) {
 }
 
 func (o *BinaryOperation) execute(span Span) (Static, error) {
+	recording := o.b.Recording
+	if recording {
```

Comment:
This code is crafted to stay out of the hot-path as much as possible.  There may be a way to improve it even further.  However I can say that previously pulling LHS.execute into a function pointer made things slower.